test/api_saveProduct_test.dart

```diff
@@ -487,4 +488,19 @@ void main() {
         // some tests can be slow here
         Duration(seconds: 90),
       ));
+
+  test('No nutrition data with nutriments', () async {
```

Comment:
As said earlier, I don't believe in maintaining `hasData`, therefore I don't think we should send an exception in that case - we should just ignore the nutriment parameter over which noNutritionData has priority.

What could be tested instead is:
* create a product with nutriments
* get the product: nutriments, and noNutritionData not true
* save same product with noNutritionData = true
* get the product: null nutriments, and noNutritionData true